source/extensions/filters/network/http_connection_manager/config.cc

```diff
@@ -498,6 +498,74 @@ const Network::Address::Instance& HttpConnectionManagerConfig::localAddress() {
   return *context_.localInfo().address();
 }
 
+// Singleton registration via macro defined in envoy/singleton/manager.h
+SINGLETON_MANAGER_REGISTRATION(hcm_date_provider);
+SINGLETON_MANAGER_REGISTRATION(hcm_route_config_provider_manager);
+SINGLETON_MANAGER_REGISTRATION(hcm_scoped_routes_config_provider_manager);
+
+// TODO(junr03): some of this code can be shared with the other factory code. Clean up
+// if this factory approach is well received by reviewers.
```

Comment:
Yeah this seems fine, so would like to see the cleaned up shared version

Reply:
updated